pkg/cmd/pull_request/create/create.go

```diff
@@ -120,3 +147,24 @@ func (o *Options) Run() error {
 
 	return nil
 }
+
+func updateNecessary(head string, base string, updateAllowed bool, scmClient *scm.Client, ctx context.Context, fullName string) (bool, int) {
+	if !updateAllowed {
+		return false, 0
+	}
+	pullRequestListOptions := scm.PullRequestListOptions{Page: 1, Size: 5000, Open: true, Closed: false}
```

Comment:
Why 5000? Bit bucket for example sets the max at 100, so anything greater than that will result in bad request.
See: https://github.com/jenkins-x/go-scm/pull/319/files#diff-d11bfd98e6b72330250844f3d1b2a7b94e88a4245870e9eb8bfcb58e2bef3918R48

Reply:
> Why 5000?
I wanted a value that could reasonably assure that it wouldn't miss open PRs, particularly as the use case for this feature is an automated system (which could create many pull requests).

> Bit bucket for example sets the max at 100
That is great context on constraints. Will noodle.

Reply:
Curious, should it be a configurable parameter from the cli? That way, users of different scm providers can set whichever value they want (according to the scm limits)

Reply:
> Curious, should it be a configurable parameter from the cli? That way, users of different scm providers can set whichever value they want (according to the scm limits)

I don't think any user would understand why they would pass a page and size flag to a command to create a pull request.

Indeed, this make me wonder if the go-scm API is correct here. As a consumer of go-scm I expect to receive a slice of pull request objects. I would understand passing a length for that list (e.g. "Please provide me the most recent 50 or 500 PRs"). But questions of page or size limits should be the concern of the code within go-scm. Should the [PullRequestListOptions](https://github.com/jenkins-x/go-scm/blob/3b877743ecdbe91790294fdf6592e2b3c85380a2/scm/pr.go#L73) be changed?

Alternatively, should the provider specific code for listing PRs in go-scm handle `page` and `size` values of 0, setting them to reasonable defaults for the given provider?